config/routes.rb

```diff
@@ -124,7 +124,7 @@
         defaults format: :json do
           resources :applications, path: "npq-applications", only: %i[index show], param: :ecf_id do
             post :reject, path: "reject"
-            post :accept, path: "accept"
+            post :accept, path: "accept", on: :member
```

Comment:
I'm not sure `on: :member` is necessary here.

It just changes the route method from `accept_api_v3_application` to `api_v3_application_accept` and the param name from `:ecf_id` to `:application_ecf_id` - it doesn't actually affect the route itself.

Reply:
I think putting these actions inside `member` is the correct way as its a singular action on a resource.
Would prefer a block though:
```ruby
member do
  post :reject, path: "reject"
  post :accept, path: "accept"
end
```

Reply:
It's still a member route by default though, adding `member` just changes the id param from `:ecf_id` to `:application_ecf_id`.

Reply:
Happy either way really, but if we use `:member` it should apply to both.

Reply:
I think naming the id param as `:ecf_id` makes more sense as we have ecf_id in the line above for the index/show actions, so we reuse the same strong params method across all endpoints in the controller as well.

```resources :applications, path: "npq-applications", only: %i[index show], param: :ecf_id do```

```      
def application_params
   params.permit(:ecf_id, filter: %i[cohort updated_since])
end
```